packages/components/src/components/LinkingProvider.tsx

```diff
@@ -0,0 +1,67 @@
+/**
+ * TODO: a preface about how and why the linking is handled.
+ *
+ * To add a new link type, follow the 4 required steps defined below.
+ */
+import React from "react";
+
+// STEP 1 of adding new link definitions:
+// If the new link uses any props not already listed here, add them.
+export interface ThunderstoreLinkProps {
+  community?: string;
+  package?: string;
+}
+
+// STEP 2 of adding new link definitions:
+// If any new properties were added to ThunderstoreLinkProps, add a
+// dummy value for them here (the value doesn't matter). This is used to
+// strip the props so they don't get passed to HTML elements in React
+// implementations. Second, redundant declaration as an object is
+// required since the stripping is done during runtime.
+export const thunderstoreLinkProps: ThunderstoreLinkProps = {
+  community: "",
+  package: "",
+};
+
+// Accepting any and all props is required to keep the linking
+// implementation-agnostic. E.g. an implementation with Chakra has all
+// kinds of custom properties.
+interface AnyProps {
+  [key: string]: string | number | boolean;
```

Comment:
`object` not needed?

Reply:
Moreover why not just `any`?

Reply:
I live in constant fear of my immortal soul being eternally punished for invoking the forbidden `any`.